### DISCUSSION TITLE
Command 'Dev Containers: Rebuild Container' intermittently fails

### DISCUSSION THREAD

**Discussion by alexanderilyin** [devcontainers/community#154] [General]
```
Version: 1.92.2 (user setup)
Commit: fee1edb8d6d72a0ddff41e5f71a671c23ed924b9
Date: 2024-08-14T17:29:30.058Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.19045
```

```
[2024-08-22T10:47:59.858Z] Stop (76 ms): Run: docker rm -f 8c28408c9d9c4b163905901fb07bf647a946af67b8293ef6f0cce4f2a184acbb
[2024-08-22T10:47:59.861Z] Command failed: docker rm -f 8c28408c9d9c4b163905901fb07bf647a946af67b8293ef6f0cce4f2a184acbb
[2024-08-22T10:47:59.861Z] Error response from daemon: removal of container 8c28408c9d9c4b163905901fb07bf647a946af67b8293ef6f0cce4f2a184acbb is already in progress
[2024-08-22T10:47:59.861Z] Exit code 1
```

I can't find a repo with related sources to put a PR, could someone please point me?

**Comment by bamurtaugh** [comment]
Thanks for opening. Looping in @chrmarti as it seems this is Dev Containers extension related.

**Reply by bamurtaugh** [reply]
Additionally @alexanderilyin, could you please share some more details on your scenario so that we can try to reproduce? i.e. could you share your dev container config or a link to a repo using your config / a similar one?

**Reply by chrmarti** [reply]
We are tracking this with https://github.com/microsoft/vscode-remote-release/issues/6509.

**Comment by alexanderilyin** [comment]
Got reproduce again:

1. Add new extension to devcontainer.json.
2. Rebuild container via command palette.

![image](https://github.com/user-attachments/assets/a3694719-9e11-4e78-863c-72c81e9811bd)

```
> code --version 
1.94.2
384ff7382de624fb94dbaf6da11977bba1ecd427
x64
```

```
> code --list-extensions --show-versions
1password.op-vscode@1.0.5
42crunch.vscode-openapi@4.29.2
bbenoist.vagrant@0.5.0
bierner.markdown-checkbox@0.4.0
bierner.markdown-footnotes@0.1.1
bierner.markdown-mermaid@1.25.0
bpruitt-goddard.mermaid-markdown-syntax-highlighting@1.6.6
codeium.codeium@1.22.5
dbaeumer.vscode-eslint@3.0.10
felixfbecker.php-intellisense@2.3.14
github.codespaces@1.17.3
github.github-vscode-theme@6.3.5
github.remotehub@0.64.0
hashicorp.terraform@2.33.0
ms-azuretools.vscode-docker@1.29.3
ms-kubernetes-tools.vscode-kubernetes-tools@1.3.18
ms-python.debugpy@2024.12.0
ms-python.isort@2023.10.1
ms-python.python@2024.16.1
ms-python.vscode-pylance@2024.10.1
ms-toolsai.jupyter@2024.9.1
ms-toolsai.jupyter-keymap@1.1.2
ms-toolsai.jupyter-renderers@1.0.19
ms-toolsai.vscode-jupyter-cell-tags@0.1.9
ms-toolsai.vscode-jupyter-slideshow@0.1.6
ms-vscode-remote.remote-containers@0.388.0
ms-vscode-remote.remote-wsl@0.88.4
ms-vscode.azure-repos@0.40.0
ms-vscode.makefile-tools@0.11.13
ms-vscode.remote-repositories@0.42.0
redhat.vscode-yaml@1.15.0
shopify.ruby-lsp@0.8.8
xdebug.php-debug@1.35.0
xdebug.php-pack@1.0.3
zobo.php-intellisense@1.3.3
```

<details><summary>Dev Containers Log</summary>
<p>
<pre>
[54 ms] Dev Containers 0.388.0 in VS Code 1.94.2 (384ff7382de624fb94dbaf6da11977bba1ecd427).
[53 ms] Start: Resolving Remote
[107 ms] Setting up container for folder or workspace: c:\Users\alexa\OneDrive\Documents\GitHub\deep-space-cartel\amp
[107 ms] Context: desktop-linux
[109 ms] Start: Run: wsl -l -v
[281 ms] Start: Run: wsl -d docker-desktop-data -e /bin/sh -c echo ~
[2891 ms] Could not connect to WSL.
[2891 ms] Command failed: wsl -d docker-desktop-data -e /bin/sh -c echo ~
[2891 ms] 
[2891 ms] <3>WSL (52) ERROR: CreateProcessParseCommon:711: Failed to translate C:\Users\alexa\AppData\Local\Programs\Microsoft VS Code
<3>WSL (52) ERROR: CreateProcessParseCommon:757: getpwuid(0) failed 2
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows\system32
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows\System32\Wbem
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows\System32\WindowsPowerShell\v1.0\
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Windows\System32\OpenSSH\
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\WINDOWS\system32
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\WINDOWS
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\WINDOWS\System32\Wbem
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\WINDOWS\System32\WindowsPowerShell\v1.0\
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\WINDOWS\System32\OpenSSH\
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\Git\cmd
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\HashiCorp\Vagrant\bin
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files (x86)\GnuPG\bin
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files\Docker\Docker\resources\bin
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\WINDOWS\system32\config\systemprofile\AppData\Local\Microsoft\WindowsApps
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\alexa\AppData\Local\Programs\Microsoft VS Code\bin
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Program Files (x86)\GitHub CLI\
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\alexa\AppData\Local\GitHubDesktop\bin
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\alexa\AppData\Local\Programs\Lens\resources\cli\bin
<3>WSL (52) ERROR: UtilTranslatePathList:2866: Failed to translate C:\Users\alexa\AppData\Local\Microsoft\WinGet\Links
<3>WSL (52) ERROR: CreateProcessEntryCommon:334: getpwuid(0) failed 2
<3>WSL (52) ERROR: CreateProcessEntryCommon:505: execvpe /bin/sh failed 2
<3>WSL (52) ERROR: CreateProcessEntryCommon:508: Create process not expected to return
[2899 ms] Start: Check Docker is running
[2899 ms] Start: Run: docker version
[3058 ms] Client:
[3058 ms]  Version:           27.2.0
 API version:       1.47
 Go version:        go1.21.13
 Git commit:        3ab4256
 Built:             Tue Aug 27 14:17:17 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.34.3 (170107)
 Engine:
  Version:          27.2.0
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.21.13
  Git commit:       3ab5c7d
  Built:            Tue Aug 27 14:15:15 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.20
  GitCommit:        8fc6bcff51318944179630522a095cc9dbf9f353
 runc:
  Version:          1.1.13
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
[3090 ms] Start: Run: docker volume ls -q
[3234 ms] Start: Run: docker inspect --type container e54093e7f9c1bd048df25344c1badc4afbcb19b5f1588ecf1d2ce67ebfb7f185
[4218 ms] Start: Run: docker rm -f e54093e7f9c1bd048df25344c1badc4afbcb19b5f1588ecf1d2ce67ebfb7f185
[4352 ms] Start: Run: docker events --format {{json .}} --filter container=e54093e7f9c1bd048df25344c1badc4afbcb19b5f1588ecf1d2ce67ebfb7f185 --filter event=destroy
[5384 ms] Start: Run: docker rm -f e54093e7f9c1bd048df25344c1badc4afbcb19b5f1588ecf1d2ce67ebfb7f185
[6468 ms] Start: Run: docker rm -f e54093e7f9c1bd048df25344c1badc4afbcb19b5f1588ecf1d2ce67ebfb7f185
[7560 ms] Start: Run: docker rm -f e54093e7f9c1bd048df25344c1badc4afbcb19b5f1588ecf1d2ce67ebfb7f185
[7652 ms] Command failed: docker rm -f e54093e7f9c1bd048df25344c1badc4afbcb19b5f1588ecf1d2ce67ebfb7f185
[7652 ms] Error response from daemon: removal of container e54093e7f9c1bd048df25344c1badc4afbcb19b5f1588ecf1d2ce67ebfb7f185 is already in progress
[7653 ms] Exit code 1
</pre>
</p>
</details>

**Comment by alexanderilyin** [comment]
@bamurtaugh here is devcontainer.json as well:

```jsonc
{
    // "image": "mcr.microsoft.com/devcontainers/go:1.2.0-1.23-bookworm",
    // "dock"
    "dockerComposeFile": "docker-compose.yml",
    "service": "amp",
    "runServices": [
        "mongodb",
        "swagger-ui"
    ],
    "workspaceFolder": "/workspace/amp",
    "features": {
        "ghcr.io/deep-space-cartel/devcontainers-features/starship:1": {},
        "ghcr.io/deep-space-cartel/devcontainers-features/apt:1": {
            "packages": "shellcheck npm"
        },
        "ghcr.io/deep-space-cartel/devcontainers-features/pip:1": {
            "location": "/opt/pip",
            "packages": "pre-commit"
        },
        "ghcr.io/devcontainers/features/github-cli:1": {
            "installDirectlyFromGitHubRelease": true,
            "version": "latest"
        }
    },
    // "appPort": [
    //     8080
    // ],
    // "runArgs": [
    //     "--cap-add=SYS_ADMIN",
    //     "--cap-add=SYS_PTRACE",
    //     "--security-opt=seccomp=unconfined",
    //     "--privileged"
    // ],
    "customizations": {
        "vscode": {
            "settings": {
                "makefile.configureOnOpen": false,
                "editor.fontFamily": "'Hack Nerd Font Mono'",
                "terminal.integrated.fontFamily": "'Hack Nerd Font Mono'",
                "files.insertFinalNewline": true,
                "dotfiles.repository": "deep-space-cartel/dotfiles",
                "cSpell.enableFiletypes": [
                    "*"
                ],
                "remote.extensionKind": {
                    "ms-vscode.PowerShell": [
                        "ui"
                    ],
                    "1Password.op-vscode": [
                        "ui"
                    ]
                }
            },
            "extensions": [
                "golang.go",
                "Codeium.codeium",
                "yzhang.markdown-all-in-one",
                "mongodb.mongodb-vscode",
                "42Crunch.vscode-openapi",
                "a-h.templ",
                "Postman.postman-for-vscode",
                "ryanluker.vscode-coverage-gutters",
                "DavidAnson.vscode-markdownlint",
                "ms-vscode.makefile-tools",
                // "alexkrechik.cucumberautocomplete",
                // "GitHub.vscode-pull-request-github",
                // "eamodio.gitlens",
                // "GitHub.github-vscode-theme",
                // "GitHub.remotehub",
                // "GitHub.vscode-github-actions",
                "streetsidesoftware.code-spell-checker"
            ]
        }
    }
}
```

**Reply by alexanderilyin** [reply]
Docker Version:
```
Docker Desktop 4.34.3 (170107) is currently the newest version available.
```

**Reply by chrmarti** [reply]
The removal seems to take a lot of time. We currently give up after 3 seconds (4 attempts). I can increase that to 6 seconds to minimize the chance of giving up too early.

**Reply by alexanderilyin** [reply]
For storage I have HP SSD EX950 1TB but it's almost at full capacity, so probably that's one of the problems on my side.

**Reply by alexanderilyin** [reply]
P. S. Also would this be possible to be exposed as JSON setting so user can have some control over it?

**Comment by chrmarti** [comment]
This would be very specialized setting, I'd prefer to increase the timeout some more before considering adding a setting.

The 6 seconds timeout is part of Dev Containers 0.390.0-pre-release if you want to give that try.
